From 93c9f8aa72349a4fe33e2a732100c76f3904eb80 Mon Sep 17 00:00:00 2001
From: Fabrice Fontaine <fontaine.fabrice@gmail.com>
Date: Tue, 7 Jun 2022 19:27:13 +0200
Subject: [PATCH] src/local-elf.h: fix big endian sh

Fix the following build failure with big endian sh (e.g. sh4aeb):

In file included from nlist.c:44:
nlist.c: In function '__elf_is_okay__':
local-elf.h:223:25: error: 'ELFDATA2LMSB' undeclared (first use in this function); did you mean 'ELFDATA2LSB'?
  223 | #define ELF_TARG_DATA   ELFDATA2LMSB
      |                         ^~~~~~~~~~~~

Fixes:
 - http://autobuild.buildroot.org/results/2980fb79c208454195d77383f1ece9afbd7f981b

Signed-off-by: Fabrice Fontaine <fontaine.fabrice@gmail.com>
[Upstream status:
https://gitlab.freedesktop.org/libbsd/libbsd/-/merge_requests/19]
---
 src/local-elf.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/local-elf.h b/src/local-elf.h
index 1dd3b42..8c4794f 100644
--- a/src/local-elf.h
+++ b/src/local-elf.h
@@ -230,7 +230,7 @@
 #if defined(__LITTLE_ENDIAN__)
 #define ELF_TARG_DATA	ELFDATA2LSB
 #elif defined(__BIG_ENDIAN__)
-#define ELF_TARG_DATA	ELFDATA2LMSB
+#define ELF_TARG_DATA	ELFDATA2MSB
 #else
 #error Unknown SH endianness
 #endif
-- 
2.35.1

